src/modules/roc_sndio/target_sox/roc_sndio/player.cpp

```diff
@@ -71,18 +69,41 @@ bool Player::open(const char* name, const char* type) {
         return false;
     }
 
+    roc_log(LogInfo,
+            "player:"
+            " bits=%lu out_rate=%lu in_rate=%lu ch=%lu",
+            (unsigned long)output_->encoding.bits_per_sample,
+            (unsigned long)output_->signal.rate, (unsigned long)out_signal_.rate,
+            (unsigned long)output_->signal.channels);
+
     return true;
 }
 
 void Player::stop() {
     stop_ = 1;
 }
 
+void Player::start(pipeline::IReceiver& input) {
+    input_ = &input;
+    core::Thread::start();
+}
+
+size_t Player::get_sample_rate() const {
+    if (!output_) {
+        roc_panic("player: can't get sample rate for non-open output file or device");
+    }
+    return size_t(output_->signal.rate);
+}
+
 void Player::run() {
     roc_log(LogDebug, "player: starting thread");
 
+    if (!input_) {
+        roc_panic("player: thread is started before device or output file was opened");
```

Comment:
The panic message is a bit misleading, because input_ is set in start(), not in open().

Reply:
fixed